src/modules/transaction/data_provider/models.rs

```diff
@@ -0,0 +1,60 @@
+use crate::BigDecimal;
+
+/// *Transaction*
+#[derive(sqlx::FromRow)]
+pub(crate) struct Transaction {
+    transaction_hash: String,
+    included_in_block_hash: String,
+    block_timestamp: BigDecimal,
+    signer_account_id: String,
+    signer_public_key: String,
+    receiver_account_id: String,
+    signature: String,
+    status: String,
```

Comment:
I was more in line thinking of it as if the whole transaction finished executing. if you imagine Transaction Receipt as the root and receipts as nodes represented as a tree, then a finalized Transaction represents a tree that has all leaves finalized

It would be the same thing as Querying the RPC node for **[Transaction Status](https://docs.near.org/api/rpc/transactions#transaction-status)**

If we treated **transaction status** as "the transaction conversion to Receipt was successful."  Then we force the developer to know the difference between receipts and asynchronous execution of receipts and they might be confused as to why the response they got from our API is different than what Explorer frontend says.